pkg/cmd/auth/gitcredential/helper.go

```diff
@@ -0,0 +1,117 @@
+package login
+
+import (
+	"bufio"
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/cli/cli/pkg/cmdutil"
+	"github.com/cli/cli/pkg/iostreams"
+	"github.com/spf13/cobra"
+)
+
+type config interface {
+	Get(string, string) (string, error)
+}
+
+type CredentialOptions struct {
+	IO     *iostreams.IOStreams
+	Config func() (config, error)
+
+	Operation string
+}
+
+func NewCmdCredential(f *cmdutil.Factory, runF func(*CredentialOptions) error) *cobra.Command {
+	opts := &CredentialOptions{
+		IO: f.IOStreams,
+		Config: func() (config, error) {
+			return f.Config()
+		},
+	}
+
+	cmd := &cobra.Command{
+		Use:    "git-credential",
+		Args:   cobra.ExactArgs(1),
+		Short:  "Implements git credential helper protocol",
+		Hidden: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			opts.Operation = args[0]
+
+			if runF != nil {
+				return runF(opts)
+			}
+			return helperRun(opts)
+		},
+	}
+
+	return cmd
+}
+
+func helperRun(opts *CredentialOptions) error {
+	if opts.Operation == "store" {
+		// We pretend to implement the "store" operation, but do nothing since we already have a cached token.
+		return cmdutil.SilentError
```

Comment:
Rather than a silent error here is it possible to send back a response saying that the user should use `gh auth login/refresh` to do this operation? This seems like it would only be called if a user is explicitly trying set credentials from the `git` command, right?

Reply:
For a user, that would make sense! But the user doesn't ever interact with this helper directly. Instead, git invokes it internally when `git credential` is interacted with, and the user typically doesn't even interact with `git credential` themselves. So I don't think a more helpful error would be useful here.

Furthermore, when multiple credential helpers are configured, git currently uses them in a sort of a chained sequence, taking credentials from one and storing them into all others. This is why I felt it was important to implement `store`, but also to discard any attempt to actually store credentials, since those unrelated to GitHub we don't want to store, and those for GitHub we already have.